options.c

```diff
@@ -149,6 +152,16 @@ void parse_options(int argc, char **argv)
 					error(EXIT_FAILURE, 0, "Invalid argument for option -s: %s", optarg);
 				_options.scalemode = s - scalemodes;
 				break;
+			case 'T':
+				if ((s = strrchr(optarg, ';')) != NULL) {
+					s[0] = '\0';
```

Comment:
```c
if ((s = strrchr(optarg, ';')) != NULL) {
	s[0] = '\0';
```

By the way, this is the only part related to parsing. The reversal simplified things quite a bit.